DoubleType.h

```diff
@@ -19,9 +21,17 @@ struct DoubleType
     DoubleType& multiply( double rhs );
     DoubleType& divide( double rhs );
 
+    DoubleType& pow(const IntType& i);
+    DoubleType& pow(const FloatType& f);
+    DoubleType& pow(const DoubleType& d);
+    DoubleType& pow(double d);
 
-    operator double() {return *doublePtr;}
+    //Operators for Static Conversion
+    //Const objects can only call their const member functions
+    operator double() {return *doublePtr;} 
```

Comment:
You don't need a non-const version of this, since none of your functions take a non-const version.
- [ ] remove